tests/infer/test_gradient.py

```diff
@@ -6,27 +6,74 @@
 import pytest
 import torch
 import torch.optim
-from torch.autograd import Variable
+from torch.autograd import variable
```

Comment:
hi @martinjankowiak , after #860, it is better to use `torch.tensor` everywhere ;)